Sources/LookupParser.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+
+/// A structure used to represent a template variable, and to resolve it in a given context.
+struct LookupParser {
+  private var bits = [String]()
+  private var current = ""
+  private var partialBits = [String]()
```

Comment:
In a lookup of `my[a.b].stuff[ref][ref2].here`, the parser will go through every bit, which might contain bracket syntax. So in this case, the `partialBits` would end up being:
- `my[a.b]`
- `stuff[ref][ref2]`
- `here`

It's called `bit` because the result is a bunch of `bit`s (old name, from before this PR), and `partial`, because it's still parsing the bit. Naming things is hard, and it's not like `stuffBeforeDot` is better?